src/ctapipe/io/interpolating.py

```diff
@@ -0,0 +1,253 @@
+from abc import abstractmethod
+from typing import Any
+
+import astropy.units as u
+import numpy as np
+import tables
+from astropy.time import Time
+from scipy.interpolate import interp1d
+
+from ctapipe.core import Component, traits
+
+from .astropy_helpers import read_table
+
+
+class StepFunction:
+
+    """
+    Step function Interpolator for the gain and pedestals
+
+    Parameters
+    ----------
+    values : None | np.array
+        Numpy array of the data that is to be interpolated.
+        The first dimension needs to be an index over time
+    times : None | np.array
+        Time values over which data are to be interpolated
+        need to be sorted and have same length as first dimension of values
+    """
+
+    def __init__(
+        self,
+        times,
+        values,
+        bounds_error=True,
+        fill_value="extrapolate",
+        assume_sorted=True,
+        copy=False,
+    ):
+        self.values = values
+        self.times = times
+        self.bounds_error = bounds_error
+        self.fill_value = fill_value
+
+    def __call__(self, point):
+        if point < self.times[0]:
```

Comment:
It seems you don't check for `point > self.times[-1]` here, i.e. the other out of bounds condition.

Reply:
The tables tjarks Statistics extractors will produce do not keep track of the end at which the data is to be used, so it is not clear how to define an upper bound here. An upper bound could be parsed separately as the last point is not where the interpolation should end.

Reply:
If you are talking about #2554 , the latest version does include start and end time in each chunk